modules/ROOT/pages/dw-functions.adoc

```diff
@@ -72,9 +72,36 @@ output application/json
 
 * xref:dw-url.adoc[URL (dw::core::URL)]
 
+[[function_signatures]]
+== Function Signatures
+
+Each DataWeave function in the DataWeave Reference is identified by its function signature. A function specifies a function name, zero or more parameters, and a return type. 
+
+Basic syntax of a two-parameter function signature:
+
+`function(parameterType,parameterType): returnType`
+
+For example, the signature `contains(String, String): Boolean` for the `contains` function has two parameters, each of which accepts a value of the String type. The function returns a Boolean value.
+
+Parameters in the function signature correspond, _in order_, to their named parameters, which are described in Parameters tables for each signature. For example, in xref:dw-core-functions-contains.adoc#contains2[contains(String, String): Boolean], the Parameters table indicates that `text` is the first parameter and `toSearch` is the second.
+
+Many DataWeave functions are overloaded to handle different data types. There is a unique function signature for each variant of the function. For example, xref:dw-core-functions-isempty.adoc[isEmpty] is overloaded to support an input value of an Array, String, Object, or Null type.
+
+For more information on the data types, see xref:dataweave-type-system.adoc[Type System].
+
+=== Type Parameters in Function Signatures
+
+Function signatures can contain type parameters, which are similar to generics in some programming languages. For example, the `Array<T>` in `contains(Array<T>, Any): Boolean` indicates that the array can contain elements of any type (`T`) that DataWeave supports. By contrast, an array of a specific type or types specifies the type, for example,  `Array<String>`, `Array<Number>`, or for both types `Array<String|Number>`.
+
+=== Lambdas in Function Signatures
+
+Some parameters within function signatures are lambdas that take one or more parameters of the form `variable:Type` with a transformation expression (`-> expression`). For example, the lambda `(item: T, index: Number) -> R)` is a parameter of the xref:dw-core-functions-map.adoc[map] function (`map(Array<T>, (item: T, index: Number) -> R): Array<R>`. The lambda accepts a value of any type `T` and a value of type Number, which serve as parameters to the expression `R`.
```

Comment:
```suggestion
=== Function types in Function Signatures

Some parameters within function signatures are `function types` (functions can be passed as arguments) that take one or more parameters of the form `paramenterName:Type` and has a return type (`-> ReturnType`). For example, the function type `(item: T, index: Number) -> R)` is a parameter type of the xref:dw-core-functions-map.adoc[map] function (`map(Array<T>, (item: T, index: Number) -> R): Array<R>`. The function type accepts a value of any type `T` and a value of type Number, which serve as parameters to return a value of type`R`.
```

Changes from Lambda to Function type. The lambda is a concept on runtime (just a function without a name), the Function type is a constraint on the type system, and that constraint could be met with a lambda or a named function.
Just to follow the same logic as https://docs.mulesoft.com/mule-runtime/4.3/dataweave-type-system#function-type